ArduCopter/autoyaw.cpp

```diff
@@ -197,6 +201,13 @@ float Mode::AutoYaw::yaw()
         // changes yaw to be same as when quad was armed
         return copter.initial_armed_bearing;
 
+    case AUTO_YAW_CIRCLE:
+        if (copter.circle_nav->is_active()) {
+            return copter.circle_nav->get_yaw();
+        }
+        // return the current attitude target
+        return wrap_360_cd(copter.attitude_control->get_att_target_euler_cd().z);
```

Comment:
why not copter.ahrs.yaw_sensor ?

Reply:
we're looking for the yaw target, not current yaw

Reply:
Yes, if we constantly updated the target to the current yaw then the vehicle's yaw would drift around.